noop/__tests__/__fixtures__/basic/code.js

```diff
@@ -0,0 +1,22 @@
+const globalVariable = {prop: "test"};
+
+$shouldNotMutate(["foo"]);
+const foo = (foo, other) => {
+	foo.prop = "pie";
+	other.prop = "Don't change me";
+};
+
+$shouldNotMutate(["foo"]);
+function bar (foo, other) {
+	foo.prop = "Test";
+	other.prop = "Don't change me";
+}
+
+/**
+ * This does not currently work
```

Comment:
Let's document this in a caveats section

Reply:
Let's, but not in this PR. This has existed in the codebase for some time and is unrelated to this change. FWIW,  this comment isn't accurate for the `noop` plugin, just the `plugin` one

Reply:
Ahh! Then I will break this out into another issue.